### ISSUE TITLE
Ticket System

### ISSUE THREAD

**Issue by John-Fries-J** [John-Fries-J/GeneralPurpose-discord-bot#3]
Create a toggleable ticket system with config that allows someone to enter their ticket channel ID in the config and set a value to true in the config and a ticket system start working.

**Comment by John-Fries-J** [comment]
The current ticket system is ALMOST done, Need to allow people to open a ticket if the bot restarts, either by sending the message on restart or keeping the data saved for the ticket and only using the ticket command when absolutely necisarry. 

Need to add some permissions to the command but the channels themselfs are ok

Need to update the config on the readme

Need to adjust the name of the channel once the ticket is CLOSED

Make a new log for all the commands, Rather than spamming the current logs whenever a button is pressed, disable the current global logging for the ticket folder and do its own logs in its own event.

Move the majority of the current ticket command either to new commands and call them with the ticket orrr make a new event with all of the things inside of it.

**Comment by John-Fries-J** [comment]
The current ticket system is ALMOST done, Need to allow people to open a ticket if the bot restarts, either by sending the message on restart or keeping the data saved for the ticket and only using the ticket command when absolutely necisarry. still needs doing

Move the majority of the current ticket command either to new commands and call them with the ticket orrr make a new event with all of the things inside of it. Can be done later down the line

~~Need to add some permissions to the command but the channels themselfs are ok~~ Set, Close and Delete command are ManageMessages. The ticket command itself is Administrator. Can be changed at the top of each file though.

~~Need to update the config on the readme~~ Config is now removed. To set up the bot and the roles it uses is done by the original /ticket command. The args are [role] which is the support role [channel] which is the channel the embed will be sent and [category] which is the category that the tickets will go to.


~~Need to adjust the name of the channel once the ticket is CLOSED~~ Done, Once the ticket is closed it is labled as closed-[ticket owners username]. This IS A MUST HAVE. The way channels are deleted is to check for "closed-" so editing this will cause issues. 

~~ Make a new log for all the commands, Rather than spamming the current logs whenever a button is pressed, disable the current global logging for the ticket folder and do its own logs in its own event.~~ Moving this to the logging issue I made


Pr'ing this branch and calling this feature almost done with the two things listed above still needing to be done but not a must do.
